### PR TITLE
Checkbox / RangeSlider

### PR DESCRIPTION
**Pull request by luckylwk**
- [x] Bugfix: `Checkbox` exposes full event (`name`, `type`)
- [x] Feat: Added RangeSlider
- [x] Bugfix: Icon alignment in `Input`

### CODE REVIEW

File: src/components/Form/Readme.mdx [hunk 1]
```diff
@@ -33,33 +37,92 @@ as well as FieldsWrapper which takes a JSON config object and renders the whole
...
-  <Input
-    name="phone"
-    type="tel"
-    placeholder="Phone number"
-    onChange={e => console.log(e.target.name, e.target.value)}
-    icon={<FiPhone />}
-    label="Phone Number"
-    context="danger"
-  />
+  {() => {
```
Comment (jonashaefele) [inline src/components/Form/Readme.mdx]: Great docs example!

I saw you left the default margin? Happy to remove it in `Input`, but keep it in `Fields`?

You were right about allowing the parent to do the layout here...

File: src/components/Form/Label.js [hunk 2]
```diff
@@ -19,14 +20,17 @@ Label.propTypes = {
   ...propTypes.color,
   ...propTypes.typography,
 };
+
 Label.defaultProps = {
   lineHeight: 0,
   fontFamily: 'secondary',
   fontSize: 'label',
+  fontWeight: 'bold',
```
Comment (jonashaefele) [inline src/components/Form/Label.js]: This now accidentally bolds all children of `Label` e.g. see `TextareaMD`

probably needs a tiny bit of css on `Label`, something like:
```
& * {
  font-weight: initial;
}
```

File: src/components/Form/Readme.mdx [hunk 3]
```diff
@@ -97,27 +160,74 @@ Checkbox uses [rc-checkbox](https://github.com/react-component/checkbox) under t
 
 <Props of={Checkbox} />
 
+Checkbox will expose an `event.target` with
```
Comment (jonashaefele) [inline src/components/Form/Readme.mdx]: Why not add these to the proptypes for `Checkbox`?

add a comment above `onChange: propTypes.func`
`/** onChange({ target: { name: <String>, type: 'checkbox', checked: <bool>} */`

or even 
`/** get's passed an object that emulated a standard JS form event, only event.target keys 'name', 'checked' and 'type' are passed.) */`

Currently the `Checkbox` still behaves non-standard.:

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input/checkbox

`event.target.checked` is a bool,
where as 
`event.target.value` is a string with `'on'` or `null` as its value depending on checked status. OR the value that was passed to the HTML element as an attr. If it's checked. (Didn't know that, just read up on it).

But the way we use checkboxes in fields, it expects `event.target.value` to be undefined for checkboxes...

So if you pass back `value` in the event, we'll have to fix the event handler in `Fields`

File: src/components/Form/Readme.mdx [hunk 4]
```diff
@@ -97,27 +160,74 @@ Checkbox uses [rc-checkbox](https://github.com/react-component/checkbox) under t
...
+
 <Playground>
   {() => {
-    const [isVegan, setIsVegan] = useState(false);
-    const [isDogPerson, setIsDogPerson] = useState(false);
+    const [formData, setFormData] = useState({});
+    const [lastEvent, setLastEvent] = useState({});
+    const onChange = (e) => {
+      const { name, value } = e.target;
+      setFormData({ ...formData, [name]: value })
```
Comment (jonashaefele) [inline src/components/Form/Readme.mdx]: you're not setting `lastEvent` and the event demo box stays empty....

File: src/components/Form/Readme.mdx [hunk 5]
```diff
@@ -198,6 +308,56 @@ Invoke `getSelectStyling` by passing it the current theme object, to receive con
...
+    }
+    return (
+      <Flex>
+        <Box margin="base" padding="base" maxWidth="400px">
+          <Box bg="neutral.lighter">
+            <RangeSlider
+              name="engagement"
+              value={formData.engagement}
+              onChange={onChange}
+              label="Engagement score?"
```
Comment (jonashaefele) [inline src/components/Form/Readme.mdx]: label doesn't render.

File: src/components/Form/index.js [hunk 6]
```diff
@@ -1,10 +1,12 @@
 export { default as Input } from './Input';
+export { default as Label } from './Label';
+export { default as TextareaMd } from './TextareaMd';
 // export { default as TagsInput } from './TagsInput';
```
Comment (jonashaefele) [inline src/components/Form/index.js]: Are we still using this? Or are we just using `Creatable`?

File: src/components/Image/Avatar.js [hunk 7]
```diff
@@ -10,6 +10,8 @@ const Wrapper = styled(Box)`
   ${variant({
     prop: 'size',
     variants: {
+      xsmall: { width: '12px', height: '12px' },
+      smaller: { width: '16px', height: '16px' },
```
Comment (jonashaefele) [inline src/components/Image/Avatar.js]: this is the first time anywhere where we have `small`, `smaller` and `xsmall`.
As in `smaller` does't exist anywhere else.

Do we need so many different sizes? I kinda like to keep scales as consistent as possible across different components.

File: src/components/Form/Readme.mdx (hunk 5 above)
Comment (luckylwk) [reply src/components/Form/Readme.mdx]: Yes, I think we may just remove the label there all together, it looks weird on a `RangeSlider`

File: src/components/Form/Readme.mdx (hunk 4 above)
Comment (luckylwk) [reply src/components/Form/Readme.mdx]: Oops, will fix that!

File: src/components/Image/Avatar.js (hunk 7 above)
Comment (luckylwk) [reply src/components/Image/Avatar.js]: Will remove

File: src/components/Form/Readme.mdx (hunk 1 above)
Comment (luckylwk) [reply src/components/Form/Readme.mdx]: Yes I think its fine to leave the margin, knowing we can override it. I liked your argument that if we add a few `Input` elements it should still look good.

File: src/components/Form/FieldsSortable.js [hunk 8]
```diff
@@ -2,15 +2,13 @@ import React from 'react';
...
-// Input components...
 import { DragDropContext, Droppable, Draggable } from 'react-beautiful-dnd';
 import { FiMove } from 'react-icons/fi';
 import { Box } from '../Box';
 import { renderField } from './Fields';
 
-/**
- * Wrapper for an individual field.
- */
+// ---------------------------
```
[the remark this answers is not in the document]
Comment (luckylwk) [reply src/components/Form/FieldsSortable.js]: I'll create a separate `refactor` PR after this one to add the lines where needed

File: src/components/Form/Readme.mdx (hunk 3 above)
Comment (luckylwk) [reply src/components/Form/Readme.mdx]: Great point! I'll adjust

File: src/components/Form/index.js (hunk 6 above)
Comment (luckylwk) [reply src/components/Form/index.js]: I'll have a think and set it up in a separate PR. Probably Creatable is better (I may have already used that in the component)

File: src/components/Form/Label.js (hunk 2 above)
Comment (luckylwk) [reply src/components/Form/Label.js]: Somehow GitHub hid this from me. I'll update this now

**Comment by luckylwk** [comment]
Loving my event debugger 😂 :

image

File: src/components/Form/Readme.mdx (hunk 5 above)
Comment (jonashaefele) [reply src/components/Form/Readme.mdx]: it does, kinda, guess that's why I have huge titles over it for TR... :)

**Pull request merged**